### ISSUE TITLE
Passwords wiped out & all links lost after update

### ISSUE THREAD

**Issue by Jason-DBK** [sissbruecker/linkding#900]
Once a week I pull all my docker images then stop and start them again so I’m always running current docker images. It’s a script that runs automatically.

After starting the new/freshly pulled image:
**my superuser account password is unknown** and I have to set it from the command prompt. (Documentation on the commands to create a super user and reset passwords is not easily found in documentation)
**When I do log back in all my URLs are gone**

other than having all my links vanish, I really like this app.

**Comment by sissbruecker** [comment]
Sorry to hear that. To investigate:
- What command do you use to start the container? 
- What command did you use to reset the password?
- Do you have any logs to share from starting a new container after the update?

**Comment by Jason-DBK** [comment]
Sascha,

I always have trouble getting the app logs for my docker apps. Let me know
specifically what logs you are interested in. Here are answers to your
questions.
Jason

docker compose -f yml/linkding.yml stop

docker compose -f yml/linkding.yml up -d


After starting container I need to use this command to reset my user
password:

docker exec -it linkding python manage.py changepassword joe

All of my links are lost at this point.

I regularly export all my bookmarks so I don’t loose them.

The documentation is very light on how to create the first user, how
to,reset passwords and how to control where my data file is located.

My .env file and the data directory is in the yml directory.


.env file:

Docker container name

LD_CONTAINER_NAME=linkding

# Port on the host system that the application should be published on

LD_HOST_PORT=9091

# Directory on the host system that should be mounted as data dir into the
Docker container

LD_HOST_DATA_DIR=./data

# Can be used to run linkding under a context path, for example: linkding/

# Must end with a slash `/`

LD_CONTEXT_PATH=

# Username of the initial superuser to create, leave empty to not create one

LD_SUPERUSER_NAME=jason

# Password for the initial superuser, leave empty to disable credentials
authentication and rely on proxy authentication instead

LD_SUPERUSER_PASSWORD=jason

# Option to disable background tasks

LD_DISABLE_BACKGROUND_TASKS=False

# Option to disable URL validation for bookmarks completely

LD_DISABLE_URL_VALIDATION=False

# Enables support for authentication proxies such as Authelia

LD_ENABLE_AUTH_PROXY=False

# Name of the request header that the auth proxy passes to the application
to identify the user

# See docs/Options.md for more details

LD_AUTH_PROXY_USERNAME_HEADER=

# The URL that linkding should redirect to after a logout, when using an
auth proxy

# See docs/Options.md for more details

LD_AUTH_PROXY_LOGOUT_URL=

# List of trusted origins from which to accept POST requests

# See docs/Options.md for more details

LD_CSRF_TRUSTED_ORIGINS=


———————

Linkding.yml:

services:

  linkding:

    container_name: "${LD_CONTAINER_NAME:-linkding}"

    image: sissbruecker/linkding:latest

    ports:

      - "${LD_HOST_PORT:-9090}:9090"

    env_file:

#     - .env

      - /home/pi/yml/.env

    restart: unless-stopped

#

**Comment by jasonlearst** [comment]
Hi @Jason-DBK,

It appears that your `Linkding.yml` is missing the volume section to mount a persistent storage.

```yaml
    volumes:
      - "${LD_HOST_DATA_DIR:-./data}:/etc/linkding/data"
```

See the full example here: https://github.com/sissbruecker/linkding/blob/master/docker-compose.yml

**Comment by Jason-DBK** [comment]
Yes, that line was missing. I've added it and that seems to have resolved
the issue.

Thank you
